### PR TITLE
form builder and settings

### PR DESCRIPTION
**Pull request by liquidg3**
Form builder now works hand in hand with skills to render settings forms, load their values, and save them, along with skill side validation on field blur.

- [x] Feature
- [ ] Bug
- [ ] Tech debt

### CODE REVIEW

File: packages/spruce-next-helpers/src/skillskit/index.js [hunk 1]
```diff
@@ -106,6 +30,9 @@ const skill = {
 	},
 
 	ready: function({ resetUrlTrail = false, showHeader = true } = {}) {
+		Iframes.sendMessage({
```
Comment (gcpantazis) [inline packages/spruce-next-helpers/src/skillskit/index.js]: 🤔 did you mean to leave this here? Not sending anything.

File: packages/spruce-skill/server/events/validate-settings.js [hunk 2]
```diff
@@ -3,41 +3,45 @@ const { eventError } = require('../lib/errorHandler')
...
 	try {
 		log.debug('**** validate-settings', { event: ctx.event })
-		if (
-			!ctx.event.payload ||
-			!ctx.event.payload.settings ||
-			!ctx.event.payload.userId
-		) {
+		const { auth, event } = ctx
+
+		if (!auth.User || !auth.Organization) {
```
Comment (gcpantazis) [inline packages/spruce-skill/server/events/validate-settings.js]: Is this file only used for org-level settings?

File: packages/spruce-next-helpers/src/skillskit/index.js (hunk 1 above)
Comment (liquidg3) [reply packages/spruce-next-helpers/src/skillskit/index.js]: oops, yeah, did not mean to leave that there!

File: packages/spruce-skill/server/events/validate-settings.js (hunk 2 above)
Comment (liquidg3) [reply packages/spruce-skill/server/events/validate-settings.js]: nope, added a location check there.

**Pull request merged**